### ISSUE TITLE
default order

### ISSUE THREAD

**Issue by fvogels** [rems-project/sail#382]
As a simple exercise to familiarise myself with Sail, I made a spec for a toy CPU. It compiled (e.g., `sail -c model.sail`) fine with 0.16 but gave me errors in 0.17. My mistake having `default Order dec` come _after_ the includes instead of before. There being no error/warning about it, I assumed it didn't matter where it appeared, as long as it was specified _somewhere_.

It appears Sail 0.16 assumes the order is `dec` until it encounters `default Order`, whereas Sail 0.17 assumes `inc`.  This is of course not an issue if `default Order` appears at the very top.

Below is an example of code that succeeds or fails depending on which version of Sail is being used.

```
$include <vector_dec.sail>

/* inc causes trouble in Sail 0.16, dec causes trouble in Sail 0.17 */
default Order dec

union instruction = {
  ADD : bits(5)
}

mapping instruction_mapping : instruction <-> bits(8) = {
  ADD(n) <-> n @ 0b001
}
```

To avoid confusion, it might be helpful if the compiler were to actually demand that the `default Order` appears first instead of allowing it to be placed anywhere. Of course, this assumes there are no other directives that should also appear first. An alternative would be that an error is issued if code depending on the order is encountered before the `default Order` has been specified.

**Comment by Alasdair** [comment]
Yes, I should probably make it an error to import the vector parts of the prelude without setting default Order.

Between 0.16 and 0.17 I simplified the language by making the bit vector ordering global, rather than tracking it in a super fine-grained way for every value, which is the reason for the (unintentional) change in behaviour here.

**Comment by Alasdair** [comment]
Should be fixed by #444
